system-addon/lib/TopStoriesFeed.jsm

```diff
@@ -374,43 +366,73 @@ this.TopStoriesFeed = class TopStoriesFeed {
   // Clean up campaign impression pref by removing all campaigns that are no
   // longer part of the response, and are therefore considered inactive.
   cleanUpCampaignImpressionPref() {
-    const impressions = this.readCampaignImpressionsPref();
     const campaignIds = new Set(this.spocCampaignMap.values());
+    this.cleanUpImpressionPref(id => !campaignIds.has(id), SPOC_IMPRESSION_TRACKING_PREF);
+  }
+
+  // Clean up rec impression pref by removing all stories that are no
+  // longer part of the response.
+  cleanUpTopRecImpressionPref() {
+    const activeStories = new Set(this.stories.map(s => `${s.guid}`));
+    this.cleanUpImpressionPref(id => !activeStories.has(id), REC_IMPRESSION_TRACKING_PREF);
+  }
+
+  cleanUpImpressionPref(expired, pref) {
```

Comment:
This function wants some comments, at the very least for the first argument. It takes a function, whereas `expired` sounds more like a boolean.